src/main/java/com/mitchej123/hodgepodge/core/util/ItemStackHashingStrategy.java

```diff
@@ -0,0 +1,24 @@
+package com.mitchej123.hodgepodge.core.util;
+
+import gnu.trove.strategy.HashingStrategy;
+import net.minecraft.item.ItemStack;
+
+/*
+ * Strategy to make ItemStacks Hashable
+ *  - Taken from https://github.com/hilburn/AdvancedSystemsManager/blob/master/src/main/java/advancedsystemsmanager/flow/execution/buffers/maps/ItemStackHashingStrategy.java
+ *    under the the DBaJ (Don't Be a Jerk) non-commercial care-free license. 
+ *  (c) hilburn
+ */
+public class ItemStackHashingStrategy implements HashingStrategy<ItemStack> {
+    public static final ItemStackHashingStrategy INSTANCE = new ItemStackHashingStrategy();
+    
+    @Override
+    public int computeHashCode(ItemStack stack) {
+        return stack.getItem().hashCode() ^ stack.getItemDamage() ^ (stack.hasTagCompound() ? stack.stackTagCompound.hashCode() : 0);
```

Comment:
Wouldn't it give better distribution if you e.g. multiply item damage by some large enough prime here? It's not that expensive